### PR TITLE
Add an ngeoSearch directive

### PR DESCRIPTION
**Pull request by elemoine**
This PR adds an `ngeoSearch` directive, an `ngeoCreateGeoJSONBloodhound` service, and a `search` example.

The `ngeoSearch` directive is very generic/low-level. It just provides a directive for decorating an element with `typeahead`.

The `ngeoCreateGeoJSONBloodhound` service is a function returning a Bloodhound engine that expects GeoJSON responses from the remote service, and create `ol.Feature` objects as the suggestions.

The `search` example shows how to use the `ngeoSearch` directive and the `ngeoCreateGeoJSONBloodhound` service.

Please review.

### CODE REVIEW

**Comment by elemoine** [comment]
The `search` example is available at http://erilem.net/ngeo/search/search.html.

File: src/services/creategeojsonbloodhound.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ */
+ngeo.createGeoJSONBloodhound = function(url, opt_featureProjection,
+    opt_dataProjection) {
+  var geojsonFormat = new ol.format.GeoJSON();
+  return new Bloodhound(/** @type {BloodhoundOptions} */ ({
+    remote: {
+      url: url,
+      ajax: {
+        dataType: 'jsonp'
+      },
```
Comment (gberaudo) [inline src/services/creategeojsonbloodhound.js]: We are setting the `limit` option and using a `replace` function to handle localization.
Would it be possible to update the remote `url` of a `Bloodhound` afterwards?

**Comment by gberaudo** [comment]
@elemoine, I looked at this PR and added a comment. I think @fredj will have more.

File: src/services/creategeojsonbloodhound.js (hunk 1 above)
Comment (elemoine) [inline src/services/creategeojsonbloodhound.js]: > We are setting the limit option and using a replace function to handle localization.

Yes, I see.

And your `filter` function doesn't create `ol.Feature` objects either. So this Bloodhound factory is not useful to you…

This factory is really about creating a `Bloodhound` engine with a `filter` function returning `ol.Feature` created out of GeoJSON documents. The rest should be customizable through options.

What about the following?

```js
/**
 * @params {BloodhoundOptions|string} options Bloodhound options or a URL. If a string is passed the
 *     default options are used, with this string as the `url` in the `remote` object.
 * @param {ol.proj.Projection=} opt_featureProjection Feature projection.
 * @param {ol.proj.Projection=} opt_dataProjection Data projection.
 * @return {Bloodhound} The Bloodhound object.
 */
ngeo.createGeoJSONBloodhound = function(
    options, opt_featureProjection, opt_dataProjection) {
  var bloodhoundOptions = /** @type BloodhoundOptions */ ({
    remote: {
      url: goog.isString(options) ? options : '',
      ajax: {
        dataType: 'jsonp'
      },
      filter: function(parsedResponse) {
        // …
      }
    },
    datumTokenizer: goog.nullFunction,
    queryTokenizer: Bloodhound.tokenizers.whitespace
  };
  if (!goog.isString(options)) {
    goog.object.extend(bloodhoundOptions, options);
  }
  return new Bloodhound(bloodhoundOptions);
};
```

I don't know if it's worth it. Maybe the user should just create his Bloodhound engine himself if he needs something more specific than what this factory does.

Thoughts?


> Would it be possible to update the remote url of a Bloodhound afterwards?

Not that I know.

Comment (fredj) [inline src/services/creategeojsonbloodhound.js]: > I don't know if it's worth it. Maybe the user should just create his Bloodhound engine himself if he needs something more specific than what this factory does.

The `BloodhoundOptions|string` parameter is a good comprise.

Comment (elemoine) [inline src/services/creategeojsonbloodhound.js]: Thanks. I changed it, and added an example in the fileoverview showing how to pass a Bloodhound options object.

**Comment by elemoine** [comment]
Note that the search example currently relies on the Luxembourg dev instance's search service (http://devv3.geoportail.lu/main/wsgi/fulltextsearch). Using a static file doesn't make sense in this case.

**Comment by elemoine** [comment]
Any objection against merging this? Happy to make changes if necessary.

**Comment by asaunier** [comment]
From my perspective, it is OK for merging.

**Comment by gberaudo** [comment]
Looks good to me.

**Comment by fredj** [comment]
+1

**Pull request merged**